proxy/http/HttpSM.cc

```diff
@@ -7887,10 +7907,17 @@ HttpSM::redirect_request(const char *redirect_url, const int redirect_len)
         // Cleanup of state etc.
         url_nuke_proxy_stuff(clientUrl.m_url_impl);
         url_nuke_proxy_stuff(t_state.hdr_info.client_request.m_url_cached.m_url_impl);
-        t_state.hdr_info.client_request.method_set(origMethod, origMethod_len);
+        t_state.hdr_info.client_request.method_set(origMethod, std::min(origMethod_len, 255));
```

Comment:
So long as we are being picky, can we eradicate the naked 255 here too?

Reply:
Yes. Will fix this too.